_projects/2023/sunpy/sunkit_image.md

```diff
@@ -0,0 +1,56 @@
+---
+name: sunkit-image and integration with sunpy
+desc: Replace the sphinx base theme with a newer version.
+requirements:
+ - Experience with Python.
+ - Experience with sunpy Map.
+difficulty: medium to low
+issues:
+ - https://github.com/sunpy/sunpy-sphinx-theme/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc
+mentors:
+ - nabobalis
+ - YOUR NAME HERE
+initiatives:
+ - GSOC
+project_size:
+ - 175 h
+tags:
+ - sunpy
+ - Python
+ - refactoring
+collaborating_projects:
+ - sunpy
+---
+
+#### Description
+
+In a previous GSoC project, [sunkit-image](https://github.com/sunpy/sunkit-image), a image analysis Python library for solar physics was created.
+Functions were created to take arrays and return arrays and this was a big big oversight at the time.
+The goal of the SunPy Project is to ensure interoperability between the packages and this entails being able to pass in a sunpy.map.Map into a function and getting a sunpy.map.Map out.
+
+Therefore the aim of this GSoC project is:
+
+1. Refactor each function to accept and return a sunpy.map.Map instead of a numpy array.
```

Comment:
This may not always be advantageous. For example, in the case of the time lag function, the input is always time-dependent and outputs do not always have two spatial dimensions (e.g. you can input a time series and get back out a single value for the time lag). In this case, always returning a numpy array makes the most sense.